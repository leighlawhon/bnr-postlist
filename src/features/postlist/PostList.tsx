import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch, shallowEqual } from 'react-redux';
import { fetchPosts } from './postlistSlice';
import { RootState } from '../../app/rootReducer';
import FeaturedPosts from "./FeaturedPosts";
import { SidebarPosts } from "./SidebarPosts";
import './postlist.component.scss';

export const PostContext = React.createContext({
  postsResults: {
    posts: null as any,
    error: null as any,
    loading: true,
    userIds: null as any
  },
  selectedUserID: "",
  setSelectedUserID: null as any
});

export default function PostList() {
  const dispatch = useDispatch();
  const [selectedUserID, setSelectedUserID] = useState('1')
  const { postsResults } = useSelector(
    (state: RootState) => {
      return {
        postsLoading: state.postsResults.loading,
        postsError: state.postsResults.error,
        postsResults: state.postsResults
      }
    },
    shallowEqual
  )

  useEffect(() => {
    dispatch(fetchPosts())
  }, [dispatch])

  return (
    <>
      <div className="container">
        <PostContext.Provider value={{ postsResults, selectedUserID, setSelectedUserID }}>
          <div className="featured">
            <FeaturedPosts />
          </div>
          <div className="other">
            <SidebarPosts />
          </div>
        </PostContext.Provider>
      </div>
    </>
  );


}