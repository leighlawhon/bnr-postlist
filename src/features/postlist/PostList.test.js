import React from "react";
import { getByTestId } from '@testing-library/dom'
import configureStore from 'redux-mock-store';
import PostList from "./PostList";
import { Provider } from 'react-redux'
import TestRenderer from 'react-test-renderer';
import thunk from 'redux-thunk'
import { screen, waitFor } from '@testing-library/react'

describe('My Connected React-Redux Component', () => {
  const middlewares = [thunk]

  const { act, create } = TestRenderer;
  const mockStore = configureStore(middlewares);

  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      postsResults: {
        posts: [],
        error: null,
        loading: false,
        userIds: []
      },
    });
    act(() => {
      component = create(
        <Provider store={store}>
          <PostList />
        </Provider>
      );
    });
  });

  it("should render with given state from Redux store", async () => {
    expect(await component.toJSON()).toMatchSnapshot();
  })

})