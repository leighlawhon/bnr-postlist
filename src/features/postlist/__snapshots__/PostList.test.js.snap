// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="container"
>
  <div
    className="featured"
  >
    <div>
      <h2
        data-testid="featured-author"
      >
        Featured Author: 
        <select
          onChange={[Function]}
        />
      </h2>
      <ul />
    </div>
  </div>
  <div
    className="other"
  >
    <div>
      <h2>
        Other Posts
      </h2>
      <ul />
    </div>
  </div>
</div>
`;
